ui/app/pages/swaps/swaps-gas-customization-modal/swaps-gas-customization-modal.container.js

```diff
@@ -87,6 +89,11 @@ const mapStateToProps = (state) => {
     conversionRate,
   })
 
+  const customGasLimitTooLow = new BigNumber(customGasLimit, 16).lt(
```

Comment:
To make explicit exactly what type of numbers are being compared, and not rely on type coercion when comparing

Reply:
`minimumGasLimit` is a decimal due to https://github.com/MetaMask/metamask-extension/pull/9858/files#diff-66bb9868292a4e99fdb3021414460e0f7485f8cab217eed0b1bfca9004d32110R448

Reply:
as for `customGasLimit`...

Reply:
It will equal either `swapsCustomizationModalLimit || initialGasLimit`

`swapsCustomizationModalLimit` will either equal `null` or the `action.payload` of any dispatched `swapCustomGasModalLimitEdited` action

`swapCustomGasModalLimitEdited` is dispatched only when `setSwapsCustomizationModalLimit`, as defined in this present file is called.

`setSwapsCustomizationModalLimit` is passed to `updateCustomGasLimit` of `AdvancedGasInputs`. The `AdvancedGasInputs` container does this:
`updateCustomGasLimit: (limit) => updateCustomGasLimit(decimalToHex(limit)),`

So `updateCustomGasLimit` is only ever called with a properly formatted hex.

Reply:
`initialGasLimit` comes from ViewQuote, where it is set to `maxGasLimit` which is defined as:
```
const usedGasLimitWithMultiplier = new BigNumber(gasLimitForMax, 16)
    .times(1.4, 10)
    .round(0)
    .toString(16)
  const nonCustomMaxGasLimit = hexMax(
    `0x${decimalToHex(usedQuote?.maxGas || 0)}`,
    usedGasLimitWithMultiplier,
  )
  const maxGasLimit = customMaxGas || nonCustomMaxGasLimit
```
So `nonCustomMaxGasLimit` will always be a hex. Which leaves `customMaxGas`, which is either an empty string or set when `setSwapsTxGasLimit` from the swaps controller is called. Which is called when either `setSwapsTxGasLimit` or `customSwapsGasParamsUpdated` in `actions.js` are called. The former is only ever passed an empty string. The latter is only ever called when `onSubmit` from the present file is called. Which is only ever passed `customGasLimit`.

So `initialGasLimit` will only ever be a hex string.